flink-runtime/src/test/java/org/apache/flink/runtime/highavailability/FileSystemJobResultStoreTest.java

```diff
@@ -0,0 +1,285 @@
+
+        assertThat(fileSystemJobResultStore.getDirtyResults()).hasSize(1);
+        assertThat(getCleanResultIdsFromFileSystem()).hasSize(1);
+
+        assertThat(
+                        fileSystemJobResultStore.getDirtyResults().stream()
+                                .map(JobResult::getJobId)
+                                .collect(Collectors.toList()))
+                .containsExactlyInAnyOrder(dirtyJobResultEntry.getJobId());
+    }
+
+    @Test
+    public void testDeleteOnCommit() throws IOException {
+        Path path = new Path(temporaryFolder.toURI());
+        fileSystemJobResultStore = new FileSystemJobResultStore(path.getFileSystem(), path, true);
+
+        fileSystemJobResultStore.createDirtyResult(DUMMY_JOB_RESULT_ENTRY);
+        assertThat(expectedDirtyFile(DUMMY_JOB_RESULT_ENTRY)).exists().isFile().isNotEmpty();
+
+        fileSystemJobResultStore.markResultAsClean(DUMMY_JOB_RESULT_ENTRY.getJobId());
+        assertThat(expectedDirtyFile(DUMMY_JOB_RESULT_ENTRY)).doesNotExist();
+        assertThat(expectedCleanFile(DUMMY_JOB_RESULT_ENTRY)).doesNotExist();
+    }
+
+    private static JobResultEntry createDummyJobResultEntry() {
+        return new JobResultEntry(
+                new JobResult.Builder()
+                        .applicationStatus(ApplicationStatus.SUCCEEDED)
+                        .jobId(new JobID())
+                        .netRuntime(Long.MAX_VALUE)
+                        .build());
+    }
+
+    private List<JobID> getCleanResultIdsFromFileSystem() throws IOException {
+        final List<JobID> cleanResults = new ArrayList<>();
+
+        final File[] cleanFiles =
+                temporaryFolder.listFiles((dir, name) -> !name.endsWith("_DIRTY.json"));
+        for (File cleanFile : cleanFiles) {
+            final FileSystemJobResultStore.JsonJobResultEntry entry =
+                    MAPPER.readValue(cleanFile, FileSystemJobResultStore.JsonJobResultEntry.class);
+            cleanResults.add(entry.getJobResult().getJobId());
+        }
+
+        return cleanResults;
+    }
+
+    /**
+     * Generates the expected path for a dirty entry given a job entry.
+     *
+     * @param entry The job ID to construct the expected dirty path from.
+     * @return The expected dirty file.
+     */
+    private File expectedDirtyFile(JobResultEntry entry) {
+        return new File(
+                temporaryFolder.toURI().getPath()
+                        + "/"
+                        + entry.getJobId().toString()
+                        + "_DIRTY.json");
+    }
+
+    /**
+     * Generates the expected path for a clean entry given a job entry.
+     *
+     * @param entry The job entry to construct the expected clean path from.
+     * @return The expected clean file.
+     */
+    private File expectedCleanFile(JobResultEntry entry) {
+        return new File(
+                temporaryFolder.toURI().getPath() + "/" + entry.getJobId().toString() + ".json");
```

Comment:
```suggestion
        return new File(temporaryFolder, entry.getJobId().toString() + ".json");
```
nit: shorter version